src/ZfcUser/Authentication/Adapter/AdapterChain.php

```diff
@@ -2,139 +2,85 @@
 
 namespace ZfcUser\Authentication\Adapter;
 
-use Zend\Authentication\Adapter\AdapterInterface;
-use Zend\Authentication\Result as AuthenticationResult;
-use Zend\EventManager\Event;
-use Zend\Stdlib\RequestInterface as Request;
-use Zend\Stdlib\ResponseInterface as Response;
-use ZfcBase\EventManager\EventProvider;
-use ZfcUser\Exception;
-
-class AdapterChain extends EventProvider implements AdapterInterface
+use Zend\Authentication\Adapter\AbstractAdapter;
+use Zend\Stdlib\PriorityList;
+use Zend\Authentication\Result;
+use Zend\EventManager\EventManagerAwareTrait;
+
+/**
+ * Chainable authentication adapter for Zend\Authentication
+ * 
+ * Allows multiple authentication adapters to be tried in succession, 
+ * breaking the chain on the first adapter to return a successful result
+ */
+class AdapterChain extends AbstractAdapter
 {
+    use EventManagerAwareTrait;
+    
     /**
-     * @var AdapterChainEvent
-     */
-    protected $event;
-
-    /**
-     * Returns the authentication result
-     *
-     * @return AuthenticationResult
+     * @var PriorityList
      */
-    public function authenticate()
+    protected $adapters;
+        
+    public function __construct()
     {
-        $e = $this->getEvent();
-
-        $result = new AuthenticationResult(
-            $e->getCode(),
-            $e->getIdentity(),
-            $e->getMessages()
-        );
-
-        $this->resetAdapters();
-
-        return $result;
+        $this->adapters = new PriorityList();
+        $this->adapters->isLIFO(false);
```

Comment:
That would be Zend Framework's problem: https://github.com/zendframework/zf2/blob/master/library/Zend/Stdlib/PriorityList.php#L172